lib/vm/src/trap/traphandlers.rs

```diff
@@ -38,6 +38,25 @@ static MAGIC: u8 = 0xc0;
 
 static DEFAULT_STACK_SIZE: AtomicUsize = AtomicUsize::new(1024 * 1024);
 
+// Current definition of `ucontext_t` in the `libc` crate is incorrect
+// on aarch64-apple-drawin so it's defined here with a more accurate definition.
+#[repr(C)]
+#[cfg(all(target_arch = "aarch64", target_os = "macos"))]
+struct ucontext_t {
+    uc_onstack: libc::c_int,
+    uc_sigmask: libc::sigset_t,
+    uc_stack: libc::stack_t,
+    uc_link: *mut libc::ucontext_t,
+    uc_mcsize: usize,
+    uc_mcontext: libc::mcontext_t,
+}
+
+#[cfg(any(
+    target_os = "linux",
+    all(not(target_arch = "aarch64"), target_os = "macos")
+))]
+use libc::ucontext_t;
```

Comment:
```suggestion
#[cfg(not(all(target_arch = "aarch64", target_os = "macos")))]
use libc::ucontext_t;
```
I think this might be sufficient, and doesn't exclude other operating systems, assuming wasmer even runs on e.g. FreeBSD.

Reply:
That will fail on windows.
there might be an issue with freebsd and/or android yes. Maybe I should just exclude windows and macos/aarch64.

Reply:
I see. maybe, then, you could use something like:
```
#[cfg(all(not(target_os = "windows"), not(all(target_arch = "aarch64", target_os = "macos"))))]
```
?

Reply:
Yeah, I can do that.

I was also looking at the "unix" general cfg, instead of excluding windows.

Reply:
done